test/runtests.jl

```diff
@@ -50,7 +50,9 @@ if anyerrors
     throw("Tests failed")
 end
 
-stdin = joinpath(dirname(@__FILE__), "stdin.sh")
-ENV2 = copy(ENV)
-ENV2["JULIA_HOME"] = JULIA_HOME
-run(setenv(`bash $stdin`, ENV2))
+if !is_windows()
```

Comment:
The issue with bounds is that all past versions assume `~` is parsed as a macro. If every single one of those versions is therefore broken on Julia 0.6, then we should upper bound those versions. If you try to apply a lower upper bound on DataFrames with Julia 0.6 than whatever first version gets released based on StatsModels, then you aren't going to be able to get a working copy of DataFrames. If old versions of DataFrames are able to load on 0.6 while using 0.6-compatible versions of DataArrays, just with broken formula functionality, then maybe that would be okay. Will have to test.